ceph_deploy/pkg.py

```diff
@@ -17,7 +17,15 @@ def install(args):
         )
         rlogger = logging.getLogger(hostname)
         rlogger.info('installing packages on %s' % hostname)
+        # Do not timeout on package install. If you we this command to install
+        # e.g. ceph-selinux or some other package with long post script we can
+        # easily timeout in the 5 minutes that we use as a default timeout,
+        # turning off the timeout completely for the time we run the command
+        # should make this much more safe.
+        backup_timeout = distro.conn.global_timeout
+        distro.conn.global_timeout = None
         distro.packager.install(packages)
+        distro.conn.global_timeout = backup_timeout
         distro.conn.exit()
```

Comment:
since the connection ends here, there is no need to backup the global timeout to reset it.

'global' in this context means that it will be used for all actions during a connection. `distro.conn.process.run` has a more granular timeout, which defaults to "wait for ever", the `global` takes precedence and that is why your changeset is correct in changing that (vs. the more granular one).